torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -2489,9 +2490,46 @@ static const std::vector<OperatorGeneratorArgs> opGenArgs1{
     OperatorGeneratorArgs(
         // TODO return generator object when torchscript supports RNG
         // first-class
-        TORCH_SELECTIVE_SCHEMA("aten::manual_seed(int seed) -> ()"),
+        TORCH_SELECTIVE_SCHEMA("aten::manual_seed.default_(int seed) -> ()"),
         [](Stack& stack) { at::manual_seed(pop(stack).toInt()); },
         aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA(
+            "aten::Generator(*, str? device=None, int? seed=None) -> Generator"),
+        [](Stack& stack) {
+          auto seed = pop(stack).toOptional<int64_t>();
+          auto device_str = pop(stack);
+          auto device = c10::Device(
+              device_str.isString() ? device_str.toStringRef() : "cpu");
+          push(stack, at::make_generator_for_device(device, seed));
+        },
+        aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA("aten::initial_seed(Generator self) -> int"),
+        [](Stack& stack) {
+          auto generator = pop(stack);
+          auto current_seed = generator.toGenerator().current_seed();
+          push(stack, (int64_t)current_seed);
+        },
+        aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA(
+            "aten::manual_seed.generator(Generator self, int seed) -> Generator"),
```

Comment:
I think (a) this modifies the generator and (b) the input generator and output generator reference the same object, right?

If so, I think this should be
```
aten::manual_seed.generator(Generator(a!) self, int seed) -> Generator(a!)
```

The `(a)` means that the two generators are aliasing the same memory, and the `!` means that there is mutation.

Not sure of the semantics of `seed()` and `current_seed()` - you may need to update the schemas of some of the other ops?